### ISSUE TITLE
Node version not detected correctly on WSL2

### ISSUE THREAD

**Issue by dm19** [rust-analyzer/rust-analyzer#2522]
I'm trying to install rust analyzer vscode extension from WSL2 (Ubuntu 18.04), but I get an error message about needing node > 10.x.

```
will run: code-insiders --list-extensions
Ignoring option list-extensions: not supported for code-insiders.
Error: install client

Caused by:
    Could not install the Visual Studio Code extension. Please make sure you have at least NodeJS 10.x installed and try again.
```

 however I have node v12.13.1 in my Windows path and v13.3.0 in WSL.

Full output;

```
me@DESKTOP-V0B6ATR:~/builds/rust-analyzer$ cargo xtask install
    Finished dev [unoptimized] target(s) in 0.08s
     Running `target/debug/xtask install`

will run: cargo --version
cargo 1.39.0 (1c6ec66d5 2019-09-30)


will run: cargo install --path crates/ra_lsp_server --locked --force
  Installing ra_lsp_server v0.1.0 (/home/tpb/builds/rust-analyzer/crates/ra_lsp_server)
    Updating crates.io index
    Finished release [optimized] target(s) in 1.20s
   Replacing /home/tpb/.cargo/bin/ra_lsp_server
    Replaced package `ra_lsp_server v0.1.0 (/home/tpb/builds/rust-analyzer/crates/ra_lsp_server)` with `ra_lsp_server v0.1.0 (/home/tpb/builds/rust-analyzer/crates/ra_lsp_server)` (executable `ra_lsp_server`)

will run: npm --version
6.13.1

will run: npm install
audited 451 packages in 1.517s

6 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


will run: npm run package --scripts-prepend-node-path

> ra-lsp@0.0.1 package /home/tpb/builds/rust-analyzer/editors/code
> vsce package

Executing prepublish script 'npm run vscode:prepublish'...

> ra-lsp@0.0.1 vscode:prepublish /home/tpb/builds/rust-analyzer/editors/code
> npm run compile


> ra-lsp@0.0.1 compile /home/tpb/builds/rust-analyzer/editors/code
> rollup -c && shx cp src/utils/terminateProcess.sh bundle/terminateProcess.sh


./src/extension.ts → ./bundle/extension.js...
(!) Circular dependencies
src/server.ts -> src/config.ts -> src/server.ts
src/server.ts -> src/highlighting.ts -> src/server.ts
src/utils/diagnostics/rust.ts -> src/utils/diagnostics/SuggestedFix.ts -> src/utils/diagnostics/rust.ts
created ./bundle/extension.js in 2.9s
 DONE  Packaged: /home/tpb/builds/rust-analyzer/editors/code/ra-lsp-0.0.1.vsix (8 files, 240.67KB)
 INFO
The latest version of vsce is 1.71.0 and you have 1.70.0.
Update it now: npm install -g vsce

will run: code --version

will run: code-insiders --version
1.41.0-insider
bef029a9d21904c0085c4e8ef1335d9c33a28fa4
x64

will run: code-insiders --install-extension ./ra-lsp-0.0.1.vsix --force
Ignoring option install-extension: not supported for code-insiders.

will run: code-insiders --list-extensions
Ignoring option list-extensions: not supported for code-insiders.
Error: install client

Caused by:
    Could not install the Visual Studio Code extension. Please make sure you have at least NodeJS 10.x installed and try again.
```

**Comment by lnicola** [comment]
```
will run: code-insiders --list-extensions
Ignoring option list-extensions: not supported for code-insiders.
```

That sounds like a Code (`--list-extensions` works in 1.40), not node.js problem.

**Comment by matklad** [comment]
cc https://github.com/microsoft/vscode-remote-release/issues/385

Not sure what's going on here, we probably need to either change our installation instructions, or to provide a better error message. The relevant code is here:

https://github.com/rust-analyzer/rust-analyzer/blob/ee904594af174866cfa10474fd8fd3f23580b6af/xtask/src/main.rs#L173-L234

I personally won't be able to look into this anytime soon

**Comment by dm19** [comment]
I see now I just missed the [existing instructions](https://github.com/rust-analyzer/rust-analyzer/tree/master/docs/user#vs-code-remote) for installing on vscode on a remote host. A relevant error message for someone using the wrong install method would be great though.

**Comment by lnicola** [comment]
Presumably fixed in #3389.
